### ISSUE TITLE
unpublish file: c.save is not a function

### ISSUE THREAD

**Issue by haraldschilly** [sagemathinc/cocalc#5200]
I changed the file publishing state from being public to private and closed the dialog. then I noticed this exception:

```
Uncaught (in promise) TypeError: c.save is not a function
    set_public_path https://cocalc.com/app line 1 > eval:3224
    s https://cocalc.com/app line 1 > eval:3224
    s https://cocalc.com/app line 1 > eval:3224
    a https://cocalc.com/app line 1 > eval:3224
    a https://cocalc.com/app line 1 > eval:3224
    set_public_path https://cocalc.com/app line 1 > eval:3224
    set_public_path https://cocalc.com/app line 1 > eval:3558
    save_description https://cocalc.com/app line 1 > eval:3566
    Ve https://cocalc.com/app line 1 > eval:1309
    Ve https://cocalc.com/app line 1 > eval:1309
    wr https://cocalc.com/app line 1 > eval:1309
    wr https://cocalc.com/app line 1 > eval:1309
    kr https://cocalc.com/app line 1 > eval:1309
    Rr https://cocalc.com/app line 1 > eval:1309
    Ie https://cocalc.com/app line 1 > eval:1309
    Rr https://cocalc.com/app line 1 > eval:1309
    Rr https://cocalc.com/app line 1 > eval:1309
    $t https://cocalc.com/app line 1 > eval:1309
    Jt https://cocalc.com/app line 1 > eval:1309
    unstable_runWithPriority https://cocalc.com/app line 1 > eval:1309
    Ba https://cocalc.com/app line 1 > eval:1309
    Pe https://cocalc.com/app line 1 > eval:1309
    Yt https://cocalc.com/app line 1 > eval:1309
    _wrapper https://cocalc.com/app line 1 > eval:622
app:3224:39180
    a https://cocalc.com/app line 1 > eval:3224
    set_public_path https://cocalc.com/app line 1 > eval:3224
    <anonymous> self-hosted:1176
    set_public_path https://cocalc.com/app line 1 > eval:3558
    save_description https://cocalc.com/app line 1 > eval:3566
    <anonymous> self-hosted:1176
```

**Comment by williamstein** [comment]
This is probably crashing in src/smc-webapp/project_actions.ts at `    table.save();` (currently line 2409). This seems like it could seriously impact core functionality, so I'm marking it priority high for my bugfixes list.

**Comment by williamstein** [comment]
fixed via a40866f01ab620ccb7cd4aa70c2aaafb8349829b
